### PR TITLE
Move most damage AI_BadMove checks to AI_CalcDamage

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Moves over damage dealing moves that would either fail or absorbed by opposing mon to `AI_CalcDamage`. The benefits of this change is that the AI will not calc moves that will fail anyways and avoids inaccuracies during AI calcs. 

A scenario that was very common is this: 
```
AI faces Lanturn with known ability Volt absorb
Has moves: Thunderbolt, Surf and Ice Beam
```
The current implementation will calc all moves and come to the conclusion that Thunderbold does the most damage which will increase the score by one. Now the move goes through `AI_CheckBadMoe` and concludes that the move is bad which decreases score by 10 and the final score for all moves will be:
```
Thunderbolt: 91
Surf: 100
Ice Beam: 100
```

By moving the check to `AI_CalcDamage` thunderbolt will be seen as a zero damage move and the final score will be:
```
Thunderbolt: 80
Surf: 101
Ice Beam: 100
```

### CODE REVIEW

File: src/battle_ai_util.c [hunk 1]
```diff
@@ -366,10 +366,86 @@ static inline s32 LowestRollDmg(s32 dmg)
     return dmg;
 }
 
+bool8 IsDamageMoveUsable(u32 move, u32 battlerAtk, u32 battlerDef)
```
Comment (DizzyEggg) [inline src/battle_ai_util.c:369]: bool32

File: src/battle_ai_util.c [hunk 2]
```diff
@@ -366,10 +366,86 @@ static inline s32 LowestRollDmg(s32 dmg)
...
+        battlerDefAbility = ABILITY_NONE;
+    else
+        battlerDefAbility = aiData->abilities[battlerDef];
+
+    SetTypeBeforeUsingMove(move, battlerAtk);
+    GET_MOVE_TYPE(move, moveType);
+
+    switch (battlerDefAbility)
+    {
+    case ABILITY_NONE:
```
Comment (DizzyEggg) [inline src/battle_ai_util.c:385]: I don't think a specific check for none ability is needed here

File: src/battle_ai_util.c [hunk 3]
```diff
@@ -366,10 +366,86 @@ static inline s32 LowestRollDmg(s32 dmg)
...
+    }
+
+    return FALSE;
+}
+
 s32 AI_CalcDamage(u32 move, u32 battlerAtk, u32 battlerDef, u8 *typeEffectiveness, bool32 considerZPower, u32 weather)
 {
     s32 dmg, moveType;
     uq4_12_t effectivenessMultiplier;
+    bool8 isDamageMoveUnusable = FALSE;
```
Comment (DizzyEggg) [inline src/battle_ai_util.c:448]: bool32

**Review by DizzyEggg — CHANGES_REQUESTED** [review]
Good idea with moving the check

**Comment by AlexOn1ine** [comment]
ready for re-review!

**Review by DizzyEggg — APPROVED** [review]

**Pull request merged**